src/lib/eliom_client.client.ml

```diff
@@ -536,6 +536,29 @@ module HistCache = Map.Make (struct
   let compare = compare
 end)
 
+module History = struct
+  let urls = ref HistCache.empty
+
+  let push_current_url () =
+    let page = !active_page in
+    let id = page.page_id.state_index in
+    let url = Js.to_string Dom_html.window##.location##.href in
```

Comment:
The location is not changed in the app:
https://github.com/ocsigen/eliom/blob/50e207e76cc09d0f0491be6b4eab9c14f600a77f/src/lib/eliom_client.client.ml#L671-L672